Makefile

```diff
@@ -1,58 +1,50 @@
-DEVUPJSON = '.profile |= "uneet-dev" \
-		  |.stages.staging |= (.domain = "apienroll.dev.unee-t.com" | .zone = "dev.unee-t.com") \
-		  | .actions[0].emails |= ["kai.hendry+apienrolldev@unee-t.com"] \
-		  | .lambda.vpc.subnets |= [ "subnet-0e123bd457c082cff", "subnet-0ff046ccc4e3b6281", "subnet-0e123bd457c082cff" ] \
-		  | .lambda.vpc.security_groups |= [ "sg-0b83472a34bc17400", "sg-0f4dadb564041855b" ]'
-
-
-DEMOUPJSON = '.profile |= "uneet-demo" \
-		  |.stages.staging |= (.domain = "apienroll.demo.unee-t.com" | .zone = "demo.unee-t.com") \
-		  | .actions[0].emails |= ["kai.hendry+apienrolldemo@unee-t.com"] \
-		  | .lambda.vpc.subnets |= [ "subnet-0bdef9ce0d0e2f596", "subnet-091e5c7d98cd80c0d", "subnet-0fbf1eb8af1ca56e3" ] \
-		  | .lambda.vpc.security_groups |= [ "sg-6f66d316" ]'
-
-PRODUPJSON = '.profile |= "uneet-prod" \
-		  |.stages.production |= (.domain = "apienroll.unee-t.com" | .zone = "unee-t.com") \
-		  | .actions[0].emails |= ["kai.hendry+apienrollprod@unee-t.com"] \
-		  | .lambda.vpc.subnets |= [ "subnet-0df289b6d96447a84", "subnet-0e41c71ad02ee7e99", "subnet-01cb9ee064743ac56" ] \
-		  | .lambda.vpc.security_groups |= [ "sg-9f5b5ef8" ]'
-
-NAME=apienroll
-REPO=uneet/$(NAME)
-
-all:
-	go build
+# This script is called by the `deploy.sh` file in (this folder)
+# We use this to deploy the environments with Travis CI
+
+# We create a function to simplify getting variables for aws parameter store.
+
+define ssm
+$(shell aws --profile $(TRAVIS_PROFILE) ssm get-parameters --names $1 --with-decryption --query Parameters[0].Value --output text)
```

Comment:
What's the default TRAVIS_PROFILE?

Why is it called TRAVIS_PROFILE and not.... AWS_PROFILE?

Reply:
> What's the default TRAVIS_PROFILE?

See the code in the `travis.yml` file on line 46, 53 and 61

> Why is it called TRAVIS_PROFILE and not.... AWS_PROFILE?

Because that variable is NOT coming from AWS, that variable is defined when `travis.yml` runs and because its value will be based on the Travis CI settings for this environment.

Reply:
> What's the default TRAVIS_PROFILE?

See the code in the `travis.yml` file on line 46, 53 and 61

> Why is it called TRAVIS_PROFILE and not.... AWS_PROFILE?

Because that variable is NOT coming from AWS, that variable is defined when `travis.yml` runs and because its value will be based on the Travis CI settings for this environment.

Reply:
The idea with a Makefile and most other robust software is that you define a default that can be overridden.

These are defaults: https://github.com/Unee-T-INS/lambda2sqs/blob/master/Makefile#L1-L5 and the style you should emulate here, else when you run `make` it won't work.

Reply:
> 
> 
> The idea with a Makefile and most other robust software is that you define a default that can be overridden.
> 
> These are defaults: https://github.com/Unee-T-INS/lambda2sqs/blob/master/Makefile#L1-L5 and the style you should emulate here, else when you run `make` it won't work.

I understand the idea but we've not been able to implement that using variables instaed of hardcoded values.
In your example: https://github.com/Unee-T-INS/lambda2sqs/blob/master/Makefile#L1-L5
We need to replace the hardcoded values 
PROJECT = lambda2sqs <--- this is OK  since this is the hardcoded name of the project
STACK_NAME ?= $(PROJECT) <--- this is OK (it's a variable)
AWS_REGION = ap-southeast-1 <--- this does NOT seem to be OK: it should be a variable
DEPLOY_S3_PREFIX = lambda2sqs <--- this is OK  since this is a hardcoded prefix related to the name of the project (question: why not use the existing PROJECT variable since the value is the same?)
AWS_PROFILE = ins-dev <--- this does NOT seem to be OK: it should be a variable

Reply:
These **are variables** that can be overridden. 

DEPLOY_S3_PREFIX can be the PROJECT variable.

Reply:
> These are variables that can be overridden.

I'm sure this is true but **you've never explained HOW they can ACTUALLY be overridden**...
If you can do that (show HOW they can be overridden) then I think we'll have made huge progress.

Reply:
Sorry I thought I showed in the call / screen share.
```

[hendry@t480s m]$ make
foobar
[hendry@t480s m]$ make A_VARIABLE=somethingelse
somethingelse
[hendry@t480s m]$ cat Makefile
A_VARIABLE := foobar

all:
        @echo $(A_VARIABLE)
```